docs/decex/staking_arithmetic_dust.md

```diff
@@ -0,0 +1,153 @@
+
+### Deposit: stake-to-share conversion
+
+Within a given epoch, assuming there are `n` nominators deposited `stake_0, .., stake_n` respectively. So:
+```rust
+// For individual nominator:
+for stake_i in  stake_0, .., stake_n {
+    nominator.balance -= stake_i
+    nominator_i.shares += stake_to_share(stake_i) - D_i
+}
+
+// For the operator pool:
+deposits_in_epoch = sum(stake_0, .., stake_n)
+
+operator.current_total_stake += deposits_in_epoch
+operator.current_total_shares += stake_to_share(deposits_in_epoch) - D
+```
+
+The update of the deposited stake on both the individual nominator and the operator pool are pure addition with no arithmetic dust. While the update of the share is based on the stake-to-share conversion, and the only connection between the nominator share and the `operator.current_total_shares` is the same share price that they use to convert deposited stake.
+
+For the `INVARIANT_2` to hold:
+```rust
+stake_to_share(deposits_in_epoch) - D >= sum(stake_to_share(stake_i)) - D_i)
+
+stake_to_share(deposits_in_epoch) - D >= sum(stake_to_share(stake_i)) - sum(D_i)
+
+-D >= -sum(D_i)
+
+D <= sum(D_i)
+```
+
+So the sum of the dust of all the individual nominator's stake-to-share conversion must be larger or equal to the dust of the operator's stake-to-share conversion, otherwise, `INVARIANT_2` will be broken. For `D <= sum(D_i)`, the share `sum(D_i) - D` is left in the pool that no one is entitled to withdraw and is given to the treasury after all nominators are unlocked.
+
+### Withdraw: share-to-stake conversion
+
+Within a given epoch, assuming there are `n` nominators withdraw `share_0, .., share_n` respectively:
+
+```rust
+// For individual nominator:
+for share_i in  share_0, .., share_n {
+    nominator.shares -= share_i;
+    nominator.balance += share_to_stake(share_i) - D_i
+}
+
+// For the operator pool:
+withdrawals_in_epoch = sum(share_0, .., share_n)
+
+operator.current_total_shares -= withdrawals_in_epoch
+operator.current_total_stake -= share_to_stake(withdrawals_in_epoch) - D
+```
+
+The update of the withdrawn share on both the individual nominator and the operator pool are pure subtraction with no arithmetic dust. While the update of the stake is based on the share-to-stake conversion, and the only connection between the nominator unlocked stake and the `operator.current_total_stake` is the same share price that they use to convert withdrawn shares.
+
+For the `INVARIANT_3` to hold:
+```rust
+share_to_stake(withdrawals_in_epoch) - D >= sum(share_to_stake(share_i)) - D_i)
+
+share_to_stake(withdrawals_in_epoch) - D >= sum(share_to_stake(share_i)) - sum(D_i)
+
+-D >= -sum(D_i)
+
+D <= sum(D_i)
+```
+
+So the sum of the dust of all the individual nominator's share-to-stake conversion must be larger or equal to the dust of the operator's share-to-stake conversion, otherwise, `INVARIANT_3` will be broken. For `D <= sum(D_i)`, the stake `sum(D_i) - D` is burn.
```

Comment:
```suggestion
So the sum of the dust of all the individual nominator's share-to-stake conversion must be larger or equal to the dust of the operator's share-to-stake conversion, otherwise, `INVARIANT_3` will be broken. For `D <= sum(D_i)`, the stake `sum(D_i) - D` is burnt.
```